public/class-amazonpolly-public.php

```diff
@@ -103,76 +103,78 @@ public function content_filter( $content ) {
 		$translate_options = '';
 
 
-		// Check if Amazon Polly is enabled in WP, if yes then...
-		if ( $common->is_polly_enabled() ) {
-
-			// Check if Amazon Polly is enabled for specific post.
-			if ( get_post_meta( $post_id, 'amazon_polly_enable', true ) === '1' ) {
-
-				$audio_location    = get_post_meta( $post_id, 'amazon_polly_audio_link_location', true );
-				$selected_autoplay = get_option( 'amazon_polly_autoplay' );
-				$player_label      = get_option( 'amazon_polly_player_label' );
-
-				// Checks if this is single post view and if there is autoplay options selected.
-				if ( is_singular() && ! empty( $selected_autoplay ) ) {
-					$autoplay = 'autoplay';
-				} else {
-					$autoplay = '';
-				}
-
-
-				// Prepare "Power By" label.
-				$voice_by_part = '';
-				if ( $common->is_poweredby_enabled() ) {
-					if ( is_singular() ) {
-						$image  = __('<img src="https://d12ee1u74lotna.cloudfront.net/images/Voiced_by_Amazon_Polly_EN.png" width="100" alt="Voiced by Amazon Polly" >', $this->plugin_name);
-					/**
-					 * Filters the voiced by Polly image HTML
-					 *
-					 * @param string $image Voiced by Polly image HTML
-					 * @param string $locale The current page locale
-					 */
-					$image  = apply_filters('amazon_polly_voiced_by_html', $image, get_locale());
-            $voice_by_part = '<a href="https://aws.amazon.com/polly/" target="_blank" rel="noopener noreferrer">' . $image . '</a>';
-					}
-				}
-
-				// Removing Amazon Polly special tags.
-				$content = $content;
-				$content = preg_replace( '/-AMAZONPOLLY-ONLYAUDIO-START-[\S\s]*?-AMAZONPOLLY-ONLYAUDIO-END-/', '', $content );
-				$content = str_replace( '-AMAZONPOLLY-ONLYWORDS-START-', '', $content );
-				$content = str_replace( '-AMAZONPOLLY-ONLYWORDS-END-', '', $content );
-
-				// Create player area.
-				if ( is_singular() ) {
-
-					// By default we will show default player.
-					$audio_part = $this->include_audio_player( 'src', $audio_location, $autoplay );
-
-					// Checks if Translate functionaliy is turned on.
-					if ($common->is_translation_enabled() ) {
-						// Checks if other than default langue is choosen.
-						if(isset($_GET['amazonai-language'])) {
-
-							// Retrievie selected language.
-							$selected_language = $_GET['amazonai-language'];
-
-							if ( $source_language != $selected_language ) {
-
-								$audio_part = '';
-								foreach ($common->get_all_polly_languages() as $language_code) {
-									if ($language_code === $selected_language) {
-										$audio_part = $this->include_audio_player( $selected_language, $audio_location, $autoplay );
-									}
-								}
-							}
-						}
-					}
-
-					$subscribe_part = $this->get_subscribe_part();
-
-
-					$polly_content = '
+        // Check if Amazon Polly is enabled in WP, if yes then...
+        if(($common->is_trinity_connected() && $common->is_trinity_enough_credits($post_id)) || !$common->is_trinity_connected()){
```

Comment:
this is an exit condition - you added it and had to indent a huge pile of code. 
can you make it nicer ?

Reply:
it is not an exit condition. if the condition is true then the code continues to execute.

Reply:
doesn't matter if it's true or false, the facts that it's all or nothing makes it an exit condition

Reply:
i don't how to change it. there is still code after the condition that should run any way.

Reply:
extract the condition to a method, and add it to the lower one `if ($common->is_polly_enabled() && trinity_enabled())`